src/codegen/codegen_c_visitor.cpp

```diff
@@ -1131,20 +1119,6 @@ bool CodegenCVisitor::nrn_cur_reduction_loop_required() {
 }
 
 
-/**
- * \details For CPU backend we iterate over all node counts.
- */
-void CodegenCVisitor::print_channel_iteration_block_begin(BlockType type) {
```

Comment:
I would not remove those - they can be useful in another case and they dont' hurt.

Reply:
@pramodk I told me that you prefer them to be remove. Let's debate (I don't care)

Reply:
For context - this `iteration block begin` was added at the beginning for **CPU OpenMP** backend where we were generating code with nested for loops. The idea was we can do nested-openmp parallelism i.e. once at top cell levels (NrnThreads) and then at individual mechanism levels (all instances of particular mechanism). This was never used/implemented.

I think it's a good idea to remove this now because 1) we removed OpenMP CPU backend months ago 2) if we will do fine-grained parallelism then (most likely) it will be more like multi-split that neuron does today 3) doing parallelism at the mechanism is a bit more complicated as we have to handle synapses sharing the same location or ion accumulation. And then `#pragma omp atomic` inside `#prgma simd / ivdep` doesn't work 😒.

Considering these aspects and simplifying the code, I suggested removing these empty functions. In case we will implement such a thing, it would be easy to bring back.

Reply:
OK, makes sense (we did use them also in the ISPC context), but as you said we don't have any use right now and they are unlikely to be used in the future - I just thought that it's just a couple lines of code that are generic enough that they "might" be useful.
Let's remove!